File: frb_example/pure_dart/lib/src/rust/api/primitive_misc.dart
// ignore_for_file: invalid_use_of_internal_member, unused_import

import '../frb_generated.dart';
import 'package:flutter_rust_bridge/flutter_rust_bridge_for_generated.dart';

Future<int> primitiveTypes(
        {required int myI32,
        required BigInt myI64,
        required double myF64,
        required bool myBool,
        dynamic hint}) =>
    RustLib.instance.api.primitiveTypes(
        myI32: myI32, myI64: myI64, myF64: myF64, myBool: myBool, hint: hint);

Future<int> primitiveU32({required int myU32, dynamic hint}) =>
    RustLib.instance.api.primitiveU32(myU32: myU32, hint: hint);

File: frb_example/pure_dart/rust/src/api/primitive_misc.rs
use log::info;

pub fn primitive_types_twin_normal(my_i32: i32, my_i64: i64, my_f64: f64, my_bool: bool) -> i32 {
    info!(
        "primitive_types({}, {}, {}, {})",
        my_i32, my_i64, my_f64, my_bool
    );
    42
}

pub fn primitive_u32_twin_normal(my_u32: u32) -> u32 {
    info!("primitive_u32({})", my_u32);
    assert_eq!(my_u32, 0xff112233);
    let ret = 0xfe112233;
    info!("returning {}", ret);
    ret
}
